src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -199,46 +293,27 @@ impl<'a, 'tcx> Qualifier<'a, 'tcx, 'tcx> {
         }
     }
 
-    /// Add the given qualification to self.qualif.
-    fn add(&mut self, qualif: Qualif) {
-        self.qualif = self.qualif | qualif;
-    }
-
-    /// Add the given type's qualification to self.qualif.
-    fn add_type(&mut self, ty: Ty<'tcx>) {
-        self.add(Qualif::MUTABLE_INTERIOR | Qualif::NEEDS_DROP);
-        self.qualif.restrict(ty, self.tcx, self.param_env);
-    }
-
-    /// Within the provided closure, self.qualif will start
+    /// Within the provided closure, `self.not_const` will start
     /// out empty, and its value after the closure returns will
-    /// be combined with the value before the call to nest.
+    /// be combined with the value before the call to `nest`.
     fn nest<F: FnOnce(&mut Self)>(&mut self, f: F) {
-        let original = self.qualif;
-        self.qualif = Qualif::empty();
+        let original = self.not_const;
+        self.not_const = false;
         f(self);
-        self.add(original);
+        self.not_const |= original;
```

Comment:
Pretty sure this is broken because it doesn't *actually* nest the other two.

Reply:
Well, it could work, but *only* if the other two aren't accessed from this one.

Reply:
Yep, you're right... the other two are accessed.